### PR TITLE
Fix edge cases where uploads munged display names

### LINKED ISSUE #48135: Upload CSV containing Cyrillic words with spaces and special characters fail to render
### Describe the bug

Upload CSV containing Cyrillic words with spaces and special characters fail to render.

### To Reproduce

1. Go to Upload CSV and upload the following CSV [hello.csv](https://github.com/user-attachments/files/17147106/hello.csv) ... You will notice 2 columns having a space and a special character respectively

image

2. You will notice that once you upload the csv you get a strange looking column for the two of them:

image

### Expected behavior

The proper name gets reflected in the Upload CSV

### Logs

```
{:database 26,
 :type :native,
 :native
 {:query
  "CREATE TABLE \"uploads\".\"hello_20240926110149\" (\"_mb_row_id\" BIGSERIAL, \"id\" BOOLEAN, \"%D0%BF%D0%B0%D1%80%D1%82%D0%BD%D0%B5%D1%80\" BOOLEAN, \"%D0%B0%D0%B4%D1%80%D0%B5%D1%81\" BOOLEAN, \"unnamed_column\" BOOLEAN, \"%D0%B4%D0%B0%D1%82%D0%B0%D0%B2_%D1%8B%D0%B5%D0%B7%D0%B4%D0%B0\" BOOLEAN, \"%D0%BF%D0%BE%D1%87%D0%B8%D0%BD%D0%B8%D0%BB%D0%B8_\" BOOLEAN, PRIMARY KEY(\"_mb_row_id\"))"}}

[1572d83d-bb19-4bae-9ce7-432773e62486] 2024-09-26T13:01:49+02:00 DEBUG metabase.query-processor.writeback Executing query

{:database 26,
 :type :native,
 :native
 {:query
  "CREATE TABLE \"uploads\".\"hello_20240926110149\" (\"_mb_row_id\" BIGSERIAL, \"id\" BOOLEAN, \"%D0%BF%D0%B0%D1%80%D1%82%D0%BD%D0%B5%D1%80\" BOOLEAN, \"%D0%B0%D0%B4%D1%80%D0%B5%D1%81\" BOOLEAN, \"unnamed_column\" BOOLEAN, \"%D0%B4%D0%B0%D1%82%D0%B0%D0%B2_%D1%8B%D0%B5%D0%B7%D0%B4%D0%B0\" BOOLEAN, \"%D0%BF%D0%BE%D1%87%D0%B8%D0%BD%D0%B8%D0%BB%D0%B8_\" BOOLEAN, PRIMARY KEY(\"_mb_row_id\"))"}}
```

### Information about your Metabase installation

```JSON
Tested both on 50 and stats
```

### Severity

It blocks cyrillic users from uploading files and forces them to update the file each time

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by crisptrutski**
Fixes https://github.com/metabase/metabase/issues/48135

This fixes some scenarios where we would not preserve the exact display name from the source CSV, due to the relationship between the humanization of the munged form involved more than just capitalization.

This bug affected any column with non-ascii characters and a space.

### CODE REVIEW

File: test/metabase/upload_test.clj [hunk 1]
```diff
@@ -770,21 +770,21 @@
...
-                                        "5,中村美咲,32,データサイエンティスト,札幌"]))]
+                  :file (csv-file-with ["ID,名前,年齢,職業,都市,Дтв ызд"
+                                        "1,佐藤太郎,25,エンジニア,東京,9"
+                                        "2,鈴木花子,30,デザイナー,大阪,8"
+                                        "3,田中一郎,28,マーケター,名古屋,7"
+                                        "4,山田次郎,35,プロジェクトマネージャー,福岡,6"
+                                        "5,中村美咲,32,データサイエンティスト,札幌,5"]))]
           (testing "Check the data was uploaded into the table correctly"
-            (is (= (header-with-auto-pk ["ID" "名前" "年齢" "職業" "都市"])
+            (is (= (header-with-auto-pk ["ID" "名前" "年齢" "職業" "都市" "Дтв Ызд"])
```
Comment (crisptrutski) [inline test/metabase/upload_test.clj:780]: This is a nonsense header, as there's no way to suppress our linter forbidding the characters that look like "a" and "e" within "Датав Ыезда".

File: src/metabase/upload.clj [hunk 2]
```diff
@@ -428,12 +428,16 @@
...
+               :set    {:display_name case-statement}
+               :where  [:and
+                        [:= :table_id table-id]
+                        [:in [:lower :name] (keys field->display-name)]
+                        ;; Only replace display names that have not been overridden already.
+                        [:<
+                         ;; In theory, these timestamps should be identical, and when I run the application they are.
+                         ;; In our tests, however, there is a small sub-second delay, so we add some buffer.
+                         [:- :updated_at :created_at]
+                         [:interval "5 seconds"]]]})))
```
Comment (crisptrutski) [inline src/metabase/upload.clj:440]: Fingers crossed this works with all the App DB flavors 🙈

**Review by sansolovyov — APPROVED** [review]
do you think this will fix a problem in #46309?

**Comment by crisptrutski** [comment]
> do you think this will fix a problem in #46309?

Unfortunately I don't think it's related.

I'll take a look at that next though.

**Pull request merged**